python/cugraph/subgraph_extraction/subgraph_extraction.pyx

```diff
@@ -0,0 +1,68 @@
+# Copyright (c) 2019, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from c_nvgraph cimport * 
+from c_graph cimport * 
+from libcpp cimport bool
+from libc.stdint cimport uintptr_t
+from libc.stdlib cimport calloc, malloc, free
+from libc.float cimport FLT_MAX_EXP
+import cudf
+from librmm_cffi import librmm as rmm
+import numpy as np
+
+cpdef subgraph(G, vertices):
+    """
+    Compute a clustering/partitioning of the given graph using the spectral balanced
+    cut method.
+    
+    Parameters
+    ----------
+    G : cuGraph.Graph                  
+       cuGraph graph descriptor
```

Comment:
does it work both on directed and undirected graphs? We may re-use nvgraph doc
https://docs.nvidia.com/cuda/nvgraph/index.html#function-nvgraphextractsubgraphbyvertex

Reply:
Yes, it works for both directed and undirected graphs.  It doesn't actually traverse the edges, so direction is immaterial.